tests/components/evohome/test_init.py

```diff
@@ -148,6 +126,20 @@ async def test_client_request_failure_v2(
     )
 
 
+@pytest.mark.parametrize("install", [*TEST_INSTALLS, "botched"])
+async def test_setup(
+    hass: HomeAssistant,
+    evohome: EvohomeClient,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test services after setup of evohome.
+
+    Registered services vary by the type of system.
+    """
+
+    assert hass.services.async_services_for_domain(DOMAIN).keys() == snapshot
```

Comment:
Also general note of the service setup. Ideally we add all services and make sure we have error handling to tell the user its not supported.